Arnej/wire match data layout

arnej27959 opened the pull request:
@havardpe please review
@toregge FYI

arnej27959 committed:
extra override point with MatchData

arnej27959 committed:
Wire MatchDataLayout through Searchable API

Updated the Searchable interface and all implementations to accept a
MatchDataLayout reference parameter, allowing the global layout from
Query::reserveHandles to be used throughout the blueprint creation pipeline.

This enables consistent term field handle allocation across the query tree,
particularly for WordAlternatives nodes that need to allocate handles for
their child terms.

arnej27959 committed:
GC printf statements

havardpe commented on searchlib/src/vespa/searchlib/queryeval/simple_phrase_blueprint.h:
```diff
@@ -25,14 +24,17 @@ class SimplePhraseBlueprint : public ComplexLeafBlueprint
 
     // used by create visitor
-    FieldSpec getNextChildField(const FieldSpec &outer);
+    FieldSpec getNextChildField(const FieldSpec &outer, fef::MatchDataLayout &layout) const {
```
consider moving utility function to match data layout

havardpe commented on searchlib/src/tests/attribute/searchable/attribute_weighted_set_blueprint_test.cpp:
```diff
@@ -128,7 +129,8 @@ struct WS {
         FieldSpecList fields;
         fields.add(FieldSpec(field, fieldId, handle));
-        queryeval::Blueprint::UP bp = searchable.createBlueprint(requestContext, fields, *node);
+        search::fef::MatchDataLayout mdl;
```
this pattern with an 'open wire':
should verify that mdl has not been used

havardpe reviewed (commented):
I am a bit unsure about the index warmup

arnej27959 committed:
Defer MatchData creation

This commit refactors the order of operations in the blueprint/search
creation flow to ensure MatchData is created after createBlueprint() is
called, not before.

Key changes:
- MatchDataLayout: Delete copy constructor, add explicit copy() method
- WarmupIndexCollection: Copy MatchDataLayout and move it to WarmupTask,
  defer MatchData creation until
- SimplePhraseBlueprint: Make getNextChildField static as next_child_field
- Tests: Move MatchData creation to after createBlueprint() in ~40 test files
- Tests: Add assertions that MatchDataLayout remains empty after createBlueprint
